syntax = "proto3";

package courseservice;

message CreateCourseRequest {
  string title = 1;
  string description = 2;
  int32 max_score = 3;
  int32 min_score = 4;
}

message GetCourseRequest {
  int32 course_id = 1;
}

message DeleteCourseRequest {
  int32 course_id = 1;
}

message CourseResponse {
  int32 course_id = 1;
  string title = 2;
  string description = 3;
  int32 max_score = 4;
  int32 min_score = 5;
}

message DeleteCourseResponse {
  bool success = 1;
}

service CourseService {
  rpc CreateCourse(CreateCourseRequest) returns (CourseResponse) {}
  rpc GetCourse(GetCourseRequest) returns (CourseResponse) {}
  rpc DeleteCourse(DeleteCourseRequest) returns (DeleteCourseResponse) {}
}
